src/components/dev-dot-content/mdx-components/mdx-headings/index.tsx

```diff
@@ -48,30 +49,120 @@ const HEADING_LEVELS_TO_PROPS: Record<
  *
  * Note: we likely want to explicitly destructure the props incoming
  * from MDX which we want to use. As is, we intentionally override
- * the incoming props.className, props.level, props.size, and props.weight.
+ * the incoming `className`, `level`, `size`, and `weight`, and we
+ * intentionally omit the incoming `data-text-content`.
  */
 export function makeMdxHeadingElement(level: HeadingProps['level']) {
 	const fixedClassName = classNames(s.heading, s[`h${level}`])
 	const { size, weight } = HEADING_LEVELS_TO_PROPS[level]
 
-	return function MdxHeading(props) {
-		const { children, ...rest } = props
+	return function MdxHeading({
```

Comment:
I've updated this component to support the new `remark-plugin-anchor-link-data`, which is also introduced in this PR, for use with Integrations. There are several reasons we've wanted to move away from the "inject elements" approach in our older `anchor-links` plugin. As well, we don't seem to have any need for "code list item permalinks" in Integrations, which are bundled inseparably in our older `anchor-links` plugin.

These updates ensure the `MdxHeading` component works with _either_ the old `anchor-links` or newly introduced `remark-plugin-anchor-link-data` plugins. We have a related task with more detail, at 🎟 [Write RFC for refactoring anchor-links plugin's anchor element injection](https://app.asana.com/0/1100423001970639/1202560954440296/f).